CRM/Civicase/Service/CaseCategoryMenu.php

```diff
@@ -50,65 +50,66 @@ public function createItems($caseTypeCategoryName) {
       'id' => $caseCategoryMenu['id'],
       'weight' => $casesWeight + 1,
     ]);
-    $this->createCaseCategorySubmenus($caseTypeCategoryName, $permissions, $caseCategoryMenu['id']);
+    $this->createCaseCategorySubmenus($caseTypeCategory, $permissions, $caseCategoryMenu['id']);
   }
 
   /**
    * Creates the Case Category Sub Menus.
    *
-   * @param string $caseTypeCategoryName
-   *   Case category name.
+   * @param array $caseTypeCategory
+   *   Case category data.
    * @param array $permissions
    *   Permissions.
    * @param int $caseCategoryMenuId
    *   Menu ID.
    */
-  protected function createCaseCategorySubmenus($caseTypeCategoryName, array $permissions, $caseCategoryMenuId) {
-    $labelForMenu = ucfirst(strtolower($caseTypeCategoryName));
+  protected function createCaseCategorySubmenus(array $caseTypeCategory, array $permissions, $caseCategoryMenuId) {
+    $labelForMenu = ucfirst(strtolower($caseTypeCategory['label']));
+    $singularLabelForMenu = ucfirst(strtolower($caseTypeCategory['singular_label']));
 
```

Comment:
We can just set a variable `$caseTypeCategoryName = $caseTypeCategory['name']` so we won't need to change much of the code below.